includes/emails/class-wcs-email-customer-notification.php

```diff
@@ -114,10 +89,34 @@ public function trigger( $subscription_id ) {
 			|| ! WC_Subscriptions_Email_Notifications::should_send_notification()
 			|| WCS_Action_Scheduler_Customer_Notifications::is_subscription_period_too_short( $subscription )
 		) {
-			// Update message code to display correct message.
+			// Add order/subscription note.
 			if ( is_admin() ) {
-				add_filter( 'woocommerce_redirect_order_location', [ $this, 'filter_order_redirect' ], 10, 3 );
+				// translators: %1$s: email title.
+				$order_note_msg = sprintf( __( 'Skipped sending %1$s: ', 'woocommerce-subscriptions' ), $this->title );
+
+				// Theoretically, this shouldn't happen, because no notification-related actions are included in the list if the
+				// global switch is off, but included it here for completeness.
+				if ( ! $this->is_enabled() ) {
+					$order_note_msg .= "\n";
+					$order_note_msg .= __( 'Global notification switch is off.', 'woocommerce-subscriptions' );
+				}
+
+				if ( ! $this->get_recipient() ) {
+					$order_note_msg .= "\n";
+					$order_note_msg .= __( 'Unknown recipient.', 'woocommerce-subscriptions' );
+				}
+
+				if ( ! WC_Subscriptions_Email_Notifications::should_send_notification() && $this->is_enabled() ) {
+					$order_note_msg .= "\n";
+					$order_note_msg .= __( 'Not a production site.', 'woocommerce-subscriptions' );
+				}
+
+				if ( WCS_Action_Scheduler_Customer_Notifications::is_subscription_period_too_short( $subscription ) ) {
+					$order_note_msg .= "\n";
+					$order_note_msg .= __( 'Subscription billing cycle is too short to send notifications.', 'woocommerce-subscriptions' );
+				}
 			}
+			$subscription->add_order_note( $order_note_msg );
```

Comment:
`$order_note_msg` only exists within the `is_admin()` condition so this could throw an error on non-admin requests.

I changed up this code a little and fixed this up in d84d136749b458cdf25c956c4fa8c149416d04dc

Reply:
Good catch, that line should have been inside the if.